martin/src/utils/utilities.rs

```diff
@@ -28,3 +32,39 @@ pub fn encode_brotli(data: &[u8]) -> Result<Vec<u8>, std::io::Error> {
     encoder.write_all(data)?;
     Ok(encoder.into_inner())
 }
+
+pub fn parse_base_path(path: &str) -> MartinResult<String> {
+    if !path.starts_with('/') {
+        return Err(BasePathError(path.to_string()));
+    }
+    if let Ok(uri) = path.parse::<Uri>() {
+        let mut result = uri.path();
+        while result.len() > 1 {
+            result = result.trim_end_matches('/');
+        }
+        return Ok(result.to_string());
+    }
+    Err(BasePathError(path.to_string()))
```

Comment:
wouldn't this be an infinite loop?

You don't need a loop - just do this:

```suggestion
    if let Ok(uri) = path.parse::<Uri>() {
      Ok(uri.path().trim_end_matches('/').to_string())
    } else {
      Err(BasePathError(path.to_string()))
    }
```

Reply:
Seems for "//" `trim_end_matches` would return empty string `""`..

Reply:
but if you leave it, it will break `format!("{base_path}/{}", path.source_ids)` -- because it would create two slashes